app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/ManifestDeploymentOptions.tsx

```diff
@@ -0,0 +1,153 @@
+import { module } from 'angular';
+import * as React from 'react';
+import { react2angular } from 'react2angular';
+import { cloneDeep, find, get, map, set, split } from 'lodash';
+import Select, { Option } from 'react-select';
+
+import { IAccountDetails, StageConfigField } from '@spinnaker/core';
+
+import { ManifestKindSearchService } from 'kubernetes/v2/manifest/ManifestKindSearch';
+
+export interface ITrafficManagementConfig {
+  enabled: boolean;
+  options: ITrafficManagementOptions;
+}
+
+export interface ITrafficManagementOptions {
+  namespace: string;
+  services: string[];
+  enableTraffic: boolean;
+  strategy: string;
+}
+
+export const defaultTrafficManagementConfig: ITrafficManagementConfig = {
+  enabled: false,
+  options: {
+    namespace: null,
+    services: [],
+    enableTraffic: false,
+    strategy: null,
+  },
+};
+
+export interface IManifestDeploymentOptionsProps {
+  accounts: IAccountDetails[];
+  config: ITrafficManagementConfig;
+  onConfigChange: (config: ITrafficManagementConfig) => void;
+  selectedAccount: string;
+}
+
+export interface IManifestDeploymentOptionsState {
+  services: string[];
+}
+
+export class ManifestDeploymentOptions extends React.Component<
+  IManifestDeploymentOptionsProps,
+  IManifestDeploymentOptionsState
+> {
+  public state: IManifestDeploymentOptionsState = { services: [] };
+
+  private onConfigChange = (key: string, value: any): void => {
+    const updatedConfig = cloneDeep(this.props.config);
+    set(updatedConfig, key, value);
+    this.props.onConfigChange(updatedConfig);
+  };
+
+  private fetchServices = (): void => {
+    const namespace = this.props.config.options.namespace;
+    const account = this.props.selectedAccount;
+    if (!namespace || !account) {
+      this.setState({
+        services: [],
+      });
+    }
+    ManifestKindSearchService.search('service', namespace, account).then(services => {
+      this.setState({ services: map(services, 'name') });
+    });
+  };
+
+  private getNamespaceOptions = (): Array<Option<string>> => {
+    const { accounts, selectedAccount } = this.props;
+    const selectedAccountDetails = find(accounts, a => a.name === selectedAccount);
+    const namespaces = get(selectedAccountDetails, 'namespaces', []);
+    return map(namespaces, n => ({ label: n, value: n }));
+  };
+
+  public componentDidMount() {
+    this.fetchServices();
+  }
+
+  public componentDidUpdate(prevProps: IManifestDeploymentOptionsProps) {
+    // If selected account changes, clear selected namespace.
+    if (prevProps.selectedAccount !== this.props.selectedAccount) {
+      this.onConfigChange('options.namespace', null);
+    }
+
+    // If selected namespace changes, clear selected services, and re-fetch services.
```

Comment:
Very minor nit: I'm not sure the comments here are necessary; the code below each is self-describing I think.

Reply:
You are very right, removed them!